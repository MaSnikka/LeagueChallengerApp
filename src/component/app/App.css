*{
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

.class-content{
  max-width: 1080px;
  margin-left: auto;
  margin-right: auto;
}

.challenge-image{
  width: 128px;
  height: 128px;
}