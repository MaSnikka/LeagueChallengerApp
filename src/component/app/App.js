import React, { Component, Fragment } from 'react';
import './App.css';
import '../menu/Titlebar.css';
import { Route, Switch, HashRouter, withRouter } from 'react-router-dom';
import 'bootstrap/dist/css/bootstrap.min.css';

import Index from "../index/Index";
import Home from '../home/Home';
import Register from "../register/Register";
import Login from "../login/Login";
import VerifyLeague from "../verifyLeague/VerifyLeague";
import { InstallPWA } from "./InstallPWA";
import Titlebar from "../menu/Titlebar";

class App extends Component {

    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div className="app">
                <HashRouter>
                    <div>
                        <Titlebar/>
                        <InstallPWA/>
                        <Switch>
                            <Route path="/" exact={true} component={Index}/>
                            <Route path='/home' exact={true} component={Home}/>
                            <Route path='/register' exact={true} component={Register}/>
                            <Route path='/login' exact={true} component={Login}/>
                            <Route path='/verifyLeague' exact={true} component={VerifyLeague}/>
                        </Switch>
                    </div>
                </HashRouter>
          </div>
      );
    }
}

export default App;
