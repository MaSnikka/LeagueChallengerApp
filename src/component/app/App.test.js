import React from 'react';
import { render, cleanup } from '@testing-library/react';
import App from './App';

//Cleanup garbage from testing
afterEach(cleanup);

describe("<App />", () => {
  //Makes snapshot of rendered HTML. If failed press U to update.
  it("Renders <App /> component correctly", () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});