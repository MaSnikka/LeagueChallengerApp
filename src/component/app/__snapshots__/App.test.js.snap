// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Renders <App /> component correctly 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <div
      class="content"
    >
      <div
        class="navbar "
        style="position: relative; z-index: 1000;"
      >
        <a
          class="menu-bars"
          href="/"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(255, 255, 255);"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            />
          </svg>
        </a>
      </div>
      <nav
        class="nav-menu"
      >
        <ul
          class="nav-menu-items"
        >
          <li
            class="navbar-toggle"
          >
            <a
              class="menu-bars"
              href="/"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(255, 255, 255);"
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                />
              </svg>
            </a>
          </li>
          <li
            class="nav-text"
          >
            <a
              href="/home"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(255, 255, 255);"
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M946.5 505L534.6 93.4a31.93 31.93 0 0 0-45.2 0L77.5 505c-12 12-18.8 28.3-18.8 45.3 0 35.3 28.7 64 64 64h43.4V908c0 17.7 14.3 32 32 32H448V716h112v224h265.9c17.7 0 32-14.3 32-32V614.3h43.4c17 0 33.3-6.7 45.3-18.8 24.9-25 24.9-65.5-.1-90.5z"
                />
              </svg>
              <span>
                Home
              </span>
            </a>
          </li>
          <li
            class="nav-text"
          >
            <a
              href="/Clan"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(255, 255, 255);"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13 6a3 3 0 11-6 0 3 3 0 016 0zM18 8a2 2 0 11-4 0 2 2 0 014 0zM14 15a4 4 0 00-8 0v3h8v-3zM6 8a2 2 0 11-4 0 2 2 0 014 0zM16 18v-3a5.972 5.972 0 00-.75-2.906A3.005 3.005 0 0119 15v3h-3zM4.75 12.094A5.973 5.973 0 004 15v3H1v-3a3 3 0 013.75-2.906z"
                />
              </svg>
              <span>
                Clan
              </span>
            </a>
          </li>
          <li
            class="nav-text"
          >
            <a
              href="/Settings"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(255, 255, 255);"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M413.967 276.8c1.06-6.235 1.06-13.518 1.06-20.8s-1.06-13.518-1.06-20.8l44.667-34.318c4.26-3.118 5.319-8.317 2.13-13.518L418.215 115.6c-2.129-4.164-8.507-6.235-12.767-4.164l-53.186 20.801c-10.638-8.318-23.394-15.601-36.16-20.801l-7.448-55.117c-1.06-4.154-5.319-8.318-10.638-8.318h-85.098c-5.318 0-9.577 4.164-10.637 8.318l-8.508 55.117c-12.767 5.2-24.464 12.482-36.171 20.801l-53.186-20.801c-5.319-2.071-10.638 0-12.767 4.164L49.1 187.365c-2.119 4.153-1.061 10.399 2.129 13.518L96.97 235.2c0 7.282-1.06 13.518-1.06 20.8s1.06 13.518 1.06 20.8l-44.668 34.318c-4.26 3.118-5.318 8.317-2.13 13.518L92.721 396.4c2.13 4.164 8.508 6.235 12.767 4.164l53.187-20.801c10.637 8.318 23.394 15.601 36.16 20.801l8.508 55.117c1.069 5.2 5.318 8.318 10.637 8.318h85.098c5.319 0 9.578-4.164 10.638-8.318l8.518-55.117c12.757-5.2 24.464-12.482 36.16-20.801l53.187 20.801c5.318 2.071 10.637 0 12.767-4.164l42.549-71.765c2.129-4.153 1.06-10.399-2.13-13.518l-46.8-34.317zm-158.499 52c-41.489 0-74.46-32.235-74.46-72.8s32.971-72.8 74.46-72.8 74.461 32.235 74.461 72.8-32.972 72.8-74.461 72.8z"
                />
              </svg>
              <span>
                Settings
              </span>
            </a>
          </li>
        </ul>
      </nav>
      <div>
        <h1>
          Dit is een onepager reclame
        </h1>
        <div
          class="mb-2"
        >
          <a
            href="/register"
          >
            <button
              class="btn btn-outline-primary btn-lg"
              data-testid="button-getStarted"
              type="button"
            >
              Get started!
            </button>
          </a>
          <a
            href="/login"
          >
            <button
              class="btn btn-outline-secondary btn-lg"
              type="button"
            >
              Already have an account?
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
